proposals/0052-svg-components.md

```diff
@@ -0,0 +1,247 @@
+  This template is reserved for anyone championing an already-approved proposal.
+
+  Community members who would like to propose an idea or feature should begin
+  by creating a GitHub Discussion. See the repo README.md for more info.
+
+  To use this template: create a new, empty file in the repo under `proposals/${ID}.md`.
+  Replace `${ID}` with the official accepted proposal ID, found in the GitHub Issue
+  of the accepted proposal.
+-->
+
+**If you have feedback and the feature is released as experimental, please leave it on the Stage 3 PR. Otherwise, comment on the Stage 2 issue (links below).**
+
+- Start Date: 2024-10-01
+- Reference Issues: <!-- related issues, otherwise leave empty -->
+- Implementation PR: https://github.com/withastro/astro/pull/12067
+- Stage 2 Issue: https://github.com/withastro/roadmap/issues/699
+- Stage 3 PR: <!-- related roadmap PR, leave it empty if you don't have a PR yet -->
+
+# Summary
+
+This RFC proposes adding native support for importing and rendering `.svg` files as Astro components, with optimized rendering techniques to minimize performance impact. It aims to allow `.svg` files to be treated as components that accept props and be optimized for repeated use on a page using `<symbol>` and `<use>` elements.
+
+# Example
+
+An SVG can be imported directly into an Astro component and used as a component that will only embed itself once. 
+
+```astro
+---
+import Logo from '../assets/logo.svg'
+---
+
+<!-- First usage: includes initial <symbol> -->
+<Logo width={48} height={48} />
+
+<!-- Subsequent usage(s): automatically optimized with <use> -->
+<Logo width={48} height={48} />
+```
+
+# Background & Motivation
+
+Include any useful background detail that that explains why this RFC is important.
+What are the problems that this RFC sets out to solve? Why now? Be brief!
+
+It can be useful to illustrate your RFC as a user problem in this section.
+(ex: "Users have reported that it is difficult to do X in Astro today.")
+
+# Goals
+
+- Enable importing and rendering `.svg` files as Astro components.
+- Allow users to pass props to the root `svg` element while allowing for prop overrides.
+- Inline `.svg` file directly without using optimized sprite rendering behavior.
+- Allow users to opt-in to optimized `.svg` rendering by utilizing `symbol` and `use` for repeated instances.
+- Maintain backwards compatibility with existing `.svg` import behavior.
+- Strip unnecessary attributes like `xmlns` and `version` from `.svg` files to reduce size.
+- Allow third-party plugins to hook into the `.svg` import behavior to generate their own component libraries.
+
+# Non-Goals
+
+- Out-of-scope: Advanced tree-shaking or SSR optimizations for large icon sets.
+- Out-of-scope: Advanced `.svg` file optimization such as `svgo`.
+- Future: Adding support for `.svg` files as framework components.
+- Non-goal: Adding support for `.json` icon sets such as `@iconify/json`.
+- Non-goal: SVG content analysis for accessibility.
```

Comment:
The RFC doesn't explain if `Markdown` and MDX will be supported/not supported. I believe we should state that somewhere, because I think users will ask that.

For example, our `astro:assets` works for both, so users might expect to have this feature to work with both.

Reply:
That's a great call out! I'll make sure that it covers Markdown usages as well! I believe it should

Reply:
This would require the use of MDX just like the `Image` component does. However, I was able to confirm my implementation works within markdown. I should probably add a test for this!